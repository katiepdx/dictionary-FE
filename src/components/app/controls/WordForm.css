.wordForm {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

input {
  margin-bottom: 1.2rem;
}
