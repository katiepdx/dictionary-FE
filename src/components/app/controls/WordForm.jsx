import React from 'react'
import PropTypes from 'prop-types'
import styles from './WordForm.css'

const WordForm = ({
  word,
  wordLanguage,
  wordTranslation,
  wordDefinition,
  exampleSentence,
  notes,
  onChange }) => {
  return (
    <div className={styles.formPage}>
      <section className={styles.wordForm}>
        <h2>Add a word!</h2>
        <label htmlFor="word">Word</label>
        <input
          id="word"
          type="text"
          name="word"
          value={word}
          onChange={onChange}
        />

        <label htmlFor="wordLanguage">word language</label>
        <input
          id="wordLanguage"
          type="text"
          name="wordLanguage"
          value={wordLanguage}
          onChange={onChange}
        />

        <label htmlFor="wordTranslation">word translation</label>
        <input
          id="wordTranslation"
          type="text"
          name="wordTranslation"
          value={wordTranslation}
          onChange={onChange}
        />

        <label htmlFor="wordDefinition">word definition</label>
        <input
          id="wordDefinition"
          type="text"
          name="wordDefinition"
          value={wordDefinition}
          onChange={onChange}
        />

        <label htmlFor="exampleSentence">example sentence</label>
        <input
          id="exampleSentence"
          type="text"
          name="exampleSentence"
          value={exampleSentence}
          onChange={onChange}
        />

        <label htmlFor="notes">notes</label>
        <textarea
          id="notes"
          type="text"
          name="notes"
          value={notes}
          onChange={onChange}
        />
      </section>
    </div>
  )
}

WordForm.propTypes = {
  word: PropTypes.string.isRequired,
  wordLanguage: PropTypes.string.isRequired,
  wordTranslation: PropTypes.string,
  wordDefinition: PropTypes.string.isRequired,
  exampleSentence: PropTypes.string.isRequired,
  notes: PropTypes.string,
  onChange: PropTypes.func,
  onClick: PropTypes.func
}

export default WordForm
