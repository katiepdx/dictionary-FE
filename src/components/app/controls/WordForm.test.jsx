import React from 'react'
import { render } from '@testing-library/react'
import WordForm from './WordForm'

describe('<WordForm/>', () => {
  it('should render add a word form to the page', () => {
    const { asFragment } = render(
      <WordForm
        word="word WordForm test"
        wordLanguage="wordLanguage WordForm test"
        wordTranslation="wordTranslation WordForm test"
        wordDefinition="wordDefinition WordForm test"
        exampleSentence="exampleSentence WordForm test"
        notes="notes I learned a new word!"
        onChange={() => { }}
      />)

    expect(asFragment()).toMatchSnapshot()
  })
})
