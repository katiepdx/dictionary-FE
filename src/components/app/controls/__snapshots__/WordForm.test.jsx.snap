// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordForm/> should render add a word form to the page 1`] = `
<DocumentFragment>
  <div
    class="formPage"
  >
    <section
      class="wordForm"
    >
      <h2>
        Add a word!
      </h2>
      <label
        for="word"
      >
        Word
      </label>
      <input
        id="word"
        name="word"
        type="text"
        value="word WordForm test"
      />
      <label
        for="wordLanguage"
      >
        word language
      </label>
      <input
        id="wordLanguage"
        name="wordLanguage"
        type="text"
        value="wordLanguage WordForm test"
      />
      <label
        for="wordTranslation"
      >
        word translation
      </label>
      <input
        id="wordTranslation"
        name="wordTranslation"
        type="text"
        value="wordTranslation WordForm test"
      />
      <label
        for="wordDefinition"
      >
        word definition
      </label>
      <input
        id="wordDefinition"
        name="wordDefinition"
        type="text"
        value="wordDefinition WordForm test"
      />
      <label
        for="exampleSentence"
      >
        example sentence
      </label>
      <input
        id="exampleSentence"
        name="exampleSentence"
        type="text"
        value="exampleSentence WordForm test"
      />
      <label
        for="notes"
      >
        notes
      </label>
      <textarea
        id="notes"
        name="notes"
        type="text"
      >
        notes I learned a new word!
      </textarea>
    </section>
  </div>
</DocumentFragment>
`;
